discovery/discoverer.go

```diff
@@ -0,0 +1,7 @@
+package discovery
```

Comment:
I'm not sure I understand when is a good reason to have a package then. We have samplers in a package, as they are a discrete bit of functionality that have no real need for the rest of veneur. Is that not a reason to have the discovery bits do the same?